modules/api-support-tiers.adoc

```diff
@@ -0,0 +1,130 @@
+// Module included in the following assemblies:
+//
+// * rest_api/understanding-api-support-tiers.adoc
+
+[id="api-tiers_{context}"]
+= API tiers
+
+All commercially supported APIs, components, and features are associated under one of the following support levels:
+
+[discrete]
+[id="api-tier-1_{context}"]
+== API tier 1
+APIs and application operating environments (AOEs) are stable within a major release for a minimum of 12 months or 3 minor releases from the announcement of deprecation, whichever is longer. The release that introduces a new or revised API or AOE, and the two following minor releases (n, n+1, n+2).
+
+[discrete]
+[id="api-tier-2_{context}"]
+== API tier 2
+APIs and AOEs are stable within a major release for a minimum of 9 months or 3 minor releases from the announcement of deprecation, whichever is longer. API tier 2 is the default for components in {product-title} where not identified.
+
+[discrete]
+[id="api-tier-3_{context}"]
+== API tier 3
+This level applies to languages, tools, applications, and optional Operators included with {product-title} through Operator Hub. Each component will specify a lifetime during which the API and AOE will be supported. Newer versions of language runtime specific components will attempt to be as API and AOE compatible from version to version as possible. Version to version compatibility is not guaranteed, however.
+
+Components and developer tools that receive continuous updates through the Operator Hub, referred to as Operators and operands, should be considered API tier 3.  Developers should use caution and understand how these components may change with each release. Users are encouraged to consult the compatibility guidelines documented by the component.
+
+[discrete]
+[id="api-tier-4_{context}"]
+== API tier 4
+No compatibility is provided. API and AOE can change at any point. These capabilities should not be used by applications needing long-term support.
+
+It is common practice for Operators to use custom resource definitions (CRDs) internally to accomplish a task.  These objects are not meant for use by actors external to the Operator and are intended to be hidden.  If any CRD is not meant for use by actors external to the Operator, the `operators.operatorframework.io/internal-objects` annotation in the Operators `ClusterServiceVersion` (CSV) must be specified, and that signals that the corresponding resource is internal use only.
```

Comment:
what's the effective difference between tier 3 and tier 4?  Both indicate they can change from version to version.  Tier 3 claims it'll "try" to maintain compatibility but it sounds like there's not commitment for either tier.

Reply:
tier 4 we actively discourage users from relying on